sycl/include/CL/sycl/detail/image_impl.hpp

```diff
@@ -283,7 +283,7 @@ class image_impl final : public SYCLMemObjT<AllocatorT> {
                     RT::PiEvent &OutEventToWait) override {
     void *UserPtr = InitFromUserData ? BaseT::getUserPtr() : nullptr;
 
-    RT::PiMemImageDesc Desc = getImageDesc();
+    RT::PiMemImageDesc Desc = getImageDesc((bool)UserPtr);
```

Comment:
Is this old C cast necessary? I hope there will be an implicit conversion.

Reply:
Replaced with "UserPtr != nullptr" in https://github.com/intel/llvm/pull/723 .